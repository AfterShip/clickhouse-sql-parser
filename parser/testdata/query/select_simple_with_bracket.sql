SELECT arrayConcat([1, 2], [3, 4], [5, 6]) AS res, f1["abc"] as f2
