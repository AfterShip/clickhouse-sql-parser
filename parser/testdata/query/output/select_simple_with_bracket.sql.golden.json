[
  {
    "SelectPos": 0,
    "StatementEnd": 66,
    "With": null,
    "Top": null,
    "HasDistinct": false,
    "SelectItems": [
      {
        "Expr": {
          "Name": {
            "Name": "arrayConcat",
            "QuoteType": 1,
            "NamePos": 7,
            "NameEnd": 18
          },
          "Params": {
            "LeftParenPos": 18,
            "RightParenPos": 41,
            "Items": {
              "ListPos": 19,
              "ListEnd": 40,
              "HasDistinct": false,
              "Items": [
                {
                  "Expr": {
                    "LeftBracketPos": 19,
                    "RightBracketPos": 24,
                    "Items": {
                      "ListPos": 20,
                      "ListEnd": 24,
                      "HasDistinct": false,
                      "Items": [
                        {
                          "Expr": {
                            "NumPos": 20,
                            "NumEnd": 21,
                            "Literal": "1",
                            "Base": 10
                          },
                          "Alias": null
                        },
                        {
                          "Expr": {
                            "NumPos": 23,
                            "NumEnd": 24,
                            "Literal": "2",
                            "Base": 10
                          },
                          "Alias": null
                        }
                      ]
                    }
                  },
                  "Alias": null
                },
                {
                  "Expr": {
                    "LeftBracketPos": 27,
                    "RightBracketPos": 32,
                    "Items": {
                      "ListPos": 28,
                      "ListEnd": 32,
                      "HasDistinct": false,
                      "Items": [
                        {
                          "Expr": {
                            "NumPos": 28,
                            "NumEnd": 29,
                            "Literal": "3",
                            "Base": 10
                          },
                          "Alias": null
                        },
                        {
                          "Expr": {
                            "NumPos": 31,
                            "NumEnd": 32,
                            "Literal": "4",
                            "Base": 10
                          },
                          "Alias": null
                        }
                      ]
                    }
                  },
                  "Alias": null
                },
                {
                  "Expr": {
                    "LeftBracketPos": 35,
                    "RightBracketPos": 40,
                    "Items": {
                      "ListPos": 36,
                      "ListEnd": 40,
                      "HasDistinct": false,
                      "Items": [
                        {
                          "Expr": {
                            "NumPos": 36,
                            "NumEnd": 37,
                            "Literal": "5",
                            "Base": 10
                          },
                          "Alias": null
                        },
                        {
                          "Expr": {
                            "NumPos": 39,
                            "NumEnd": 40,
                            "Literal": "6",
                            "Base": 10
                          },
                          "Alias": null
                        }
                      ]
                    }
                  },
                  "Alias": null
                }
              ]
            },
            "ColumnArgList": null
          }
        },
        "Modifiers": [],
        "Alias": {
          "Name": "res",
          "QuoteType": 1,
          "NamePos": 46,
          "NameEnd": 49
        }
      },
      {
        "Expr": {
          "Object": {
            "Name": "f1",
            "QuoteType": 1,
            "NamePos": 51,
            "NameEnd": 53
          },
          "Params": {
            "LeftBracketPos": 53,
            "RightBracketPos": 59,
            "Items": {
              "ListPos": 55,
              "ListEnd": 58,
              "HasDistinct": false,
              "Items": [
                {
                  "Expr": {
                    "Name": "abc",
                    "QuoteType": 2,
                    "NamePos": 55,
                    "NameEnd": 58
                  },
                  "Alias": null
                }
              ]
            }
          }
        },
        "Modifiers": [],
        "Alias": {
          "Name": "f2",
          "QuoteType": 1,
          "NamePos": 64,
          "NameEnd": 66
        }
      }
    ],
    "From": null,
    "ArrayJoin": null,
    "Window": null,
    "Prewhere": null,
    "Where": null,
    "GroupBy": null,
    "WithTotal": false,
    "Having": null,
    "OrderBy": null,
    "LimitBy": null,
    "Limit": null,
    "Settings": null,
    "Format": null,
    "UnionAll": null,
    "UnionDistinct": null,
    "Except": null
  }
]